src/Nethermind/Nethermind.Merge.Plugin/MergeHeaderValidator.cs

```diff
@@ -67,20 +79,50 @@ private bool ValidateTheMergeChecks(BlockHeader header)
         {
             bool validDifficulty = true, validNonce = true, validUncles = true;
             (bool IsTerminal, bool IsPostMerge) switchInfo = _poSSwitcher.GetBlockConsensusInfo(header);
+
             bool terminalTotalDifficultyChecks = ValidateTerminalTotalDifficultyChecks(header, switchInfo.IsTerminal);
+            
             if (switchInfo.IsPostMerge)
             {
                 validDifficulty = ValidateHeaderField(header, header.Difficulty, UInt256.Zero, nameof(header.Difficulty));
                 validNonce = ValidateHeaderField(header, header.Nonce, 0u, nameof(header.Nonce));
                 validUncles = ValidateHeaderField(header, header.UnclesHash, Keccak.OfAnEmptySequenceRlp, nameof(header.UnclesHash));
             }
 
-            return terminalTotalDifficultyChecks
+            bool valid = terminalTotalDifficultyChecks
                    && validDifficulty
                    && validNonce
                    && validUncles;
+
+            return valid;
         }
-        
+
+        private bool ValidateTerminalBlock(BlockHeader header)
+        {
+            bool terminalTotalDifficultyChecks = ValidateTerminalTotalDifficultyChecks(header, true);
+            bool hashMatch = ValidateHeaderField(header, header.Hash, _mergeConfig.TerminalBlockHashParsed, "terminal block hash");
```

Comment:
what if `TerminalBlockHash` was not set? Maybe `TerminalBlockHashParsed` should be `null` then instead of `Keccak.Zero` and this should have a `null` fallback?

Reply:
if null or Keccak.Zero we should skip this validation